sherpa/optmethods/optfcts.py

```diff
@@ -182,6 +182,18 @@ def _par_at_boundary(low: np.ndarray,
     return False
 
 
+def _update_reported_nfev(result: OptReturn,
```

Comment:
Does this really need to be a typed function? How often do we need it? I think essentially any function that has only one line should be inlined for readability. When I read the place where it's used `result[4]['nfev'] += nfev` tells me immediately what happens. Instead, I need to go and hunt for the `_update_reported_nfev` function ("update" doesn't imply +1, could also be some other change, so I actually have to read the function if I want to know what happens), see what that does, and then jump back to what I was reading before. Does not make the code clearer.

Reply:
It's a design choice I dislike, but reasonable people can disagree with my opinion. I'm not going to hold up this PR over a one-line function.

Personally, if I see `_update_reported_nfev` I will look for that function, find it, try to see what it does, and then I'll still have to think if it has the correct semantics in that function (i.e. does it have at least 5 elements and is the fifth element a dict), as opposed to in the place where it's inlined. Same thought process,  just more jumping. But can be done either way.